### PR TITLE
Labels, Names, and Tests

### PR DESCRIPTION
**Pull request by Snugug**
* Fix Default Label from Plugin
* Fix Object Reference Bug
* Update Name for inputs
* Require kebab case for IDs
* Clean up error messages

### CODE REVIEW

File: lib/content-types/merge.js [hunk 1]
```diff
@@ -60,6 +67,13 @@ const squish = (types) => {
         plugin.id = attribute.id;
         plugin.type = attribute.type;
 
+        // Set Default Label from Plugin Name
+        if (Object.keys(plugin.inputs).length === 1) {
+          const input = Object.keys(plugin.inputs)[0];
+
+          plugin.inputs[input].label = plugin.name;
```
Comment (scottnath) [inline lib/content-types/merge.js]: Confused on this...if this was used on [the text plugin](https://github.com/punchcard-cms/input-plugin-text/blob/master/index.js), then wouldn't it change the `label` from "Add your text" to "Text"? Wouldn't we rather it was "Add your text"?

from input plugin:
```
module.exports = {
  name: 'Text',
  description: 'An input plugin for a single text file',
  validation: {
    textValidation: validation,
  },
  inputs: {
    text: {
      validation: {
        function: 'textValidation',
        on: 'blur',
      },
      label: 'Add Your Text',
```

File: tests/content-types.js [hunk 2]
```diff
@@ -43,19 +30,177 @@ test('merged', t => {
...
+      },
+    ],
+  };
+
+  return types([testCT])
     .then(result => {
+      // console.log(util.inspect(result, false, null));
       t.is(result[0].name, 'FooRific', 'Get first content type name');
       t.is(result[0].description, 'A very foo content model.', 'Get first content type desc');
       t.is(result[0].id, 'foo-rific', 'Get first content type id');
```
Comment (scottnath) [inline tests/content-types.js]: if you want to overwrite label (see [comment above](https://github.com/punchcard-cms/content-types/pull/9/files#r64439926), you should add a `t.is` to check label too

File: lib/content-types/merge.js (hunk 1 above)
Comment (Snugug) [inline lib/content-types/merge.js]: This is the merge. If someone configured an input plugin, and there is only one input, the name they give the configuration should be used as the label if one isn't passed.

```yaml
name: Foo
attributes:
  - name: My Awesome Text Input
    type: text
```

That'll result in `My Awesome Text Input` being the label. You'll notice on line 62 that the plugin name is overridden with the attribute name.

File: tests/content-types.js (hunk 2 above)
Comment (Snugug) [inline tests/content-types.js]: I didn't update these tests. I can update these tests if you'd like

**Comment by scottnath** [comment]
Increased code coverage? You're welcome!

**Pull request merged**